indico/modules/events/schemas.py

```diff
@@ -5,12 +5,57 @@
 # modify it under the terms of the MIT License; see the
 # LICENSE file for more details.
 
-from marshmallow import fields
+import re
+
+from marshmallow import ValidationError, fields, validates_schema
 
 from indico.core.marshmallow import mm
 from indico.modules.events.models.events import Event
 from indico.modules.events.models.principals import EventPrincipal
-from indico.util.marshmallow import PrincipalPermissionList
+from indico.util.i18n import _
+from indico.util.marshmallow import PrincipalPermissionList, not_empty
+
+
+class ValidRegExField(fields.String):
+    """A valid Python regular expression."""
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        if len(value) < 2:
+            raise ValidationError(_('Empty regex'))
+        try:
+            return re.compile(value).pattern
+        except ValueError as exc:
+            raise ValidationError(str(exc))
+
+
+class AutoLinkerRuleSchema(mm.Schema):
+    """A regex -> url with placeholders mapping rule."""
+
+    regex = ValidRegExField()
+    url = fields.String(validate=not_empty)
+
+    @validates_schema(skip_on_field_errors=True)
+    def _check_url_is_well_formed(self, data, **kwargs):
+        regex = data['regex']
+        if not regex:
+            raise ValidationError(_('No regex provided'))
```

Comment:
`validate=not_empty`  on the  `ValidRegExField` above?

Reply:
In fact, the `ValidRegExField` already validates it internally so this is probably not even needed....